packages/dx-grid-core/src/plugins/integrated-filtering/computeds.js

```diff
@@ -14,88 +14,94 @@ const defaultPredicate = (value, filter) =>
   toLowerCase(value).indexOf(toLowerCase(filter.value)) > -1;
 
 const filterTree = (tree, predicate) =>
-  tree.reduce(
-    (acc, node) => {
-      if (node[NODE_CHECK]) {
-        const filteredChildren = filterTree(node.children, predicate);
-        if (filteredChildren.length > 0) {
-          acc.push({
-            ...node,
-            children: filteredChildren,
-          });
-          return acc;
-        } else if (predicate(node.root)) {
-          acc.push(node.root);
-          return acc;
-        }
-      }
-
-      if (predicate(node)) {
-        acc.push(node);
+  tree.reduce((acc, node) => {
+    if (node[NODE_CHECK]) {
+      const filteredChildren = filterTree(node.children, predicate);
+      if (filteredChildren.length > 0) {
+        acc.push({
+          ...node,
+          children: filteredChildren,
+        });
+        return acc;
+      } else if (predicate(node.root)) {
+        acc.push(node.root);
         return acc;
       }
+    }
 
+    if (predicate(node)) {
+      acc.push(node);
       return acc;
-    },
-    [],
-  );
+    }
+
+    return acc;
+  }, []);
 
-const filterHierarchicalRows = (rows, predicate, getRowLevelKey, isGroupRow) => {
+const filterHierarchicalRows = (rows, predicate, getRowLevelKey, isGroupNode) => {
   const tree = rowsToTree(rows, getRowLevelKey);
 
-  const filteredTree = filterTree(tree, (row) => {
-    if (isGroupRow(row)) {
-      if (row.collapsedRows) {
-        return row.collapsedRows.findIndex(predicate) > -1;
-      }
-      return false;
-    }
-    return predicate(row);
-  });
+  const filteredTree = filterTree(
+    tree,
+    node =>
+      (isGroupNode(node)
```

Comment:
Why node? It is a row

Reply:
"Row" has a tree structure. In common approach filter tree not have the row abstraction. Tree structure has leaves and nodes abstraction.